tests/quantization/test_range_init.py

```diff
@@ -243,6 +244,157 @@ def test_scope_overrides(self, wrap_dataloader):
+                "target_quantizer_group": "activations",
+                "target_scopes": ["NNCFNetwork/TwoConvTestModel[nncf_module]/Sequential[features]"]
+            }],
+            expected_modules_to_init={
+                "NNCFNetwork/ModuleDict[activation_quantizers]/SymmetricQuantizer"
+                "[TwoConvTestModel/Sequential[features]/Sequential[0]/NNCFConv2d[0]"
+                "/conv2d_0]": {
+                    "type": "mean_min_max",
+                    "num_init_steps": 2,
+                    "ignored_scopes": ["NNCFNetwork/TwoConvTestModel[nncf_module]/Sequential[features]"]
+                    },
+                "NNCFNetwork/TwoConvTestModel[nncf_module]/Sequential[features]"
+                "/Sequential[0]/NNCFConv2d[0]/ModuleDict[pre_ops]/UpdateInputs[1]"
+                "/SymmetricQuantizer[op]": {
+                    "type": "threesigma",
+                    "num_init_steps": 1,
+                    "target_quantizer_group": "activations",
+                    "target_scopes": ["NNCFNetwork/TwoConvTestModel[nncf_module]/Sequential[features]"]
+                    },
+                "NNCFNetwork/TwoConvTestModel[nncf_module]/Sequential[features]"
+                "/Sequential[0]/NNCFConv2d[0]/ModuleDict[pre_ops]/UpdateWeight[0]"
+                "/SymmetricQuantizer[op]": {
+                    "type": "min_max",
+                    "num_init_steps": 1,
+                    "target_quantizer_group": "weights",
+                    "target_scopes": ["NNCFNetwork/TwoConvTestModel[nncf_module]/Sequential[features]"]
+                    },
+                "NNCFNetwork/TwoConvTestModel[nncf_module]/Sequential[features]"
+                "/Sequential[1]/NNCFConv2d[0]/ModuleDict[pre_ops]/UpdateWeight[0]"
+                "/SymmetricQuantizer[op]": {
+                    "type": "min_max",
+                    "num_init_steps": 1,
+                    "target_quantizer_group": "weights",
+                    "target_scopes": ["NNCFNetwork/TwoConvTestModel[nncf_module]"
+                                      "/Sequential[features]"]
+                    }
+            })
+    ]
+
+    @staticmethod
+    @pytest.fixture(params=PER_LAYER_RANGE_INIT_TEST_CASES)
+    def per_layer_range_init_test_struct(request):
+        return request.param
+
+    def test_per_layer_range_init_with_correct_possible_config(self, wrap_dataloader, per_layer_range_init_test_struct):
+        config = create_config()
+        config['compression']['initializer']['range'] = per_layer_range_init_test_struct.range_init_config
+        data_loader = self.create_dataloader(wrap_dataloader, config)
+        config.register_extra_structs([QuantizationRangeInitArgs(data_loader)])
+        ctrl, _ = self.create_algo_and_compressed_model(config)
+        for str_scope, range_init_config in per_layer_range_init_test_struct.expected_modules_to_init.items():
+            assert ctrl.modules_to_range_init[str_scope][1] == range_init_config
+
 
```

Comment:
Should add a test that checks that the initializers are actually called the required number of times. Use "mock" functionality of pytest for this.

Reply:
Done.